### DISCUSSION TITLE
sdCard support, m5stack coreS3

### DISCUSSION THREAD

**Discussion by mhaberler** [sukesh-ak/BSP-IDF5-ESP_LCD-LVGL9#5] [Q&A]
Hi Sukesh,

we discussed this over at LovyanGFX  - I got this to work with M5Unified/M5GFX but not with LovyanGFX for a  m5stack coreS3 target - so here I would have a potential fallback even if I like LovyanGFX

it seems the coreS3 is part of the BSP, but I'm pretty fresh to esp-idf 

could you guide us through an example of adding a new target? 

I guess this involves tickling the component manager to add the board support

thanks in advance!

Michael

**Comment by sukesh-ak** [comment]
Not sure what the catch is but it says 

`The SD card is not working simultaneously with the LCD screen. We are working on a fix.`
https://github.com/espressif/esp-bsp/tree/master/bsp/m5stack_core_s3

I don't have that device to test it.

**Comment by sukesh-ak** [comment]
It should work if you don't re-initialize the SPI bus (for uSD) if TFT & SD Card are on the same SPI bus.

![image](https://github.com/sukesh-ak/BSP-IDF5-ESP_LCD-LVGL9/assets/53081362/7895283d-4353-4c82-9f8c-b296b50d4785)

https://docs.m5stack.com/en/core/CoreS3

**Reply by mhaberler** [reply]
so was my theory, and I tried that with lovyan but didnt get far
I'll have a look at the BSP approach eventually, currently the cores3 is not the top option and better variants like the LilyGo or Sunton units dont have that issue

**Reply by sukesh-ak** [reply]
It was working fine in my `ESP32-TUX` project which uses `LovyanGFX`. 
Maybe something specific to `CoreS3`?

**Reply by mhaberler** [reply]
good point
I'll revisit that

**Reply by sukesh-ak** [reply]
Did you try their Arduino sample?
https://github.com/m5stack/M5CoreS3/blob/main/examples/Basic/sdcard/sdcard.ino

**Reply by mhaberler** [reply]
no, because at some point in time M5CoreS3 was way behind what M5GFX/M5Unified offered with less hassle and it seems M5Stack was dropping the M5CoreS3 lib altogether in favor of M5Unified - which made sense

but thanks, yes I should try that
